src/docker-in-docker/install.sh

```diff
@@ -325,12 +326,17 @@ tee /usr/local/share/docker-init.sh > /dev/null \
 
 set -e
 
-AZURE_DNS_AUTO_DETECTION=$AZURE_DNS_AUTO_DETECTION
+dockerd_start="$(cat << INNEREOF
+    AZURE_DNS_AUTO_DETECTION=${AZURE_DNS_AUTO_DETECTION}
+    DOCKER_DEFAULT_ADDRESS_POOL=${DOCKER_DEFAULT_ADDRESS_POOL}
+    
+INNEREOF
+)"
```

Comment:
Would it be more straightforward to leave these as separate environment variables here:

```suggestion
AZURE_DNS_AUTO_DETECTION=${AZURE_DNS_AUTO_DETECTION}
DOCKER_DEFAULT_ADDRESS_POOL=${DOCKER_DEFAULT_ADDRESS_POOL}
```

...and then be explicit about injecting them down below like this?

```
dockerd_start="$(AZURE_DNS_AUTO_DETECTION=${AZURE_DNS_AUTO_DETECTION} DOCKER_DEFAULT_ADDRESS_POOL=${DOCKER_DEFAULT_ADDRESS_POOL} cat << 'INNEREOF'

```

Reply:
That won't work. The heredoc below is quoted so it doesn't allow variable expansion.

Reply:
Oh, nvm, I mis-interpreted what you were saying. I'm fine with that change.